File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        cnt = defaultdict(int)
        l = 0
        max_freq = 0
        ans = 0

        for r in range(len(s)):
            cnt[s[r]]+=1
            max_freq = max(max_freq, cnt[s[r]])

            while ((r-l+1)-max_freq) > k:
                cnt[s[l]] -= 1
                l+=1
            
            ans = max(ans, r-l+1) 
        return ans

File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int>ans(n+1, 0);
        int cnt = 0;
        for(int i = 0; i<=n;i++){
            cnt = 0;
            int tmp = i;
            while(tmp){
                if (tmp%2==1) cnt++;
                tmp/=2;
            }
            ans[i] = cnt;
            cnt = 0;
        }
        return ans;
    }
};
